File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/EventDAO.java
package com.jann_luellmann.thekenapp.data.dao;

import com.jann_luellmann.thekenapp.data.model.Event;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface EventDAO {

    @Query("SELECT * FROM event")
    LiveData<List<Event>> getAll();

    @Query("SELECT * FROM event WHERE id IN (:eventIds)")
    LiveData<List<Event>> loadAllByIds(long[] eventIds);

    @Query("SELECT * FROM event WHERE name LIKE :name LIMIT 1")
    LiveData<Event> findByName(String name);

    @Insert
    void insertAll(Event... events);

    @Update
    void update(Event event);

    @Delete
    void delete(Event event);
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/EventDAO.kt
package com.jann_luellmann.thekenapp.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.jann_luellmann.thekenapp.data.model.Drink
import com.jann_luellmann.thekenapp.data.model.Event

@Dao
interface EventDAO {
    @get:Query("SELECT * FROM event ORDER BY date")
    val all: LiveData<List<Event>>

    @Query("SELECT * FROM event WHERE eventId IN (:eventIds)")
    fun loadAllByIds(eventIds: LongArray): LiveData<List<Event>>

    @Query("SELECT * FROM event WHERE ename LIKE :name LIMIT 1")
    fun findByName(name: String): LiveData<Event>

    @Insert
    suspend fun insert(event: Event): Long

    @Insert
    fun insertAll(vararg events: Event)

    @Update
    fun update(event: Event)

    @Delete
    fun delete(event: Event)
}